tstest/integration/nat/nat_test.go

```diff
@@ -450,34 +453,51 @@ var types = []nodeType{
 	{"sameLAN", sameLAN},
 }
 
+// want sets the expected ping route for the test.
+func (nt *natTest) want(r pingRoute) {
+	if nt.gotRoute != r {
+		nt.tb.Errorf("ping route = %v; want %v", nt.gotRoute, r)
+	}
+}
+
 func TestEasyEasy(t *testing.T) {
 	nt := newNatTest(t)
 	nt.runTest(easy, easy)
+	nt.want(routeDirect)
 }
 
-// Tests https://github.com/tailscale/tailscale/issues/3824 ...
+func TestSameLAN(t *testing.T) {
+	nt := newNatTest(t)
+	nt.runTest(easy, sameLAN)
+	nt.want(routeLocal)
+}
+
+// TestBPFDisco tests https://github.com/tailscale/tailscale/issues/3824 ...
 // * server behind a Hard NAT
 // * client behind a NAT with UPnP support
 // * client machine has a stateful host firewall (e.g. ufw)
-
 func TestBPFDisco(t *testing.T) {
 	nt := newNatTest(t)
 	nt.runTest(easyPMPFW, hard)
+	nt.want(routeDirect)
```

Comment:
This is what we want, but this test currently fails. (not on CI, but when run manually)